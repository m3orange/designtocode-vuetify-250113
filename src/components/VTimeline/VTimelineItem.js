// Components
import VIcon from '../VIcon'

// Mixins
import Colorable from '../../mixins/colorable'

export default {
  name: 'v-timeline-item',

  mixins: [Colorable],

  props: {
    icon: {
      type: String
    },
    iconColor: {
      type: String
    },
    noIcon: {
      type: Boolean,
      default: false
    },
    iconFillColor: {
      type: String,
      default: 'white'
    }
  },

  computed: {
    iconFillClass () {
      return this.addBackgroundColorClassChecks({}, this.iconFillColor)
    }
  },

  methods: {
        genIcon () {
      const iconElement = this.noIcon
        ? null
        : this.$createElement( VIcon, { props: { color: this.iconColor } },
            this.icon )
      
      return this.$createElement('div', {
          staticClass: 'v-timeline__icon',
          class: this.iconFillClass
        }, [iconElement])
    },
    genBody () {
      return this.$createElement('div', {
          staticClass: 'v-timeline__body'
        }, this.$slots.item)
    }
  },

  render (h) {
    const children = []

    children.push(this.genIcon())
    children.push(this.genBody())

    return h('li', {
        staticClass: 'v-timeline-item'
      }, children)
  }
}
