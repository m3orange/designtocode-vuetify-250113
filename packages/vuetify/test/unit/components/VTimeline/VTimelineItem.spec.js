import { test } from '@/test'
import VTimelineItem from '@/components/VTimeline/VTimelineItem'

test('VTimelineItem.js', ({ mount }) => {
  it('should conditionally render dot', () => {
    const wrapper = mount(VTimelineItem, {
      propsData: {
        hideDot: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()

    wrapper.setProps({ hideDot: false })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should conditionally render an icon or icon slot', () => {
    expect(mount(VTimelineItem, {
      slots: {
        icon: [{
          render: h => h('div', 'foo')
        }]
      }
    }).html()).toMatchSnapshot()

    expect(mount(VTimelineItem, {
      propsData: { icon: 'foo' }
    }).html()).toMatchSnapshot()
  })

  it('should render opposite slot', () => {
    expect(mount(VTimelineItem, {
      slots: {
        opposite: [{
          render: h => h('div', 'foo')
        }]
      }
    }).html())
  })

  it('should render a small icon correctly', () => {
    expect(mount(VTimelineItem, {
      propsData: {
        small: true
      }
    }).html()).toMatchSnapshot()
  })

  it('should render a medium icon correctly', () => {
    expect(mount(VTimelineItem, {
      propsData: {
        medium: true
      }
    }).html()).toMatchSnapshot()
  })

  it('should render a large icon correctly', () => {
    expect(mount(VTimelineItem, {
      propsData: {
        large: true
      }
    }).html()).toMatchSnapshot()
  })
})
