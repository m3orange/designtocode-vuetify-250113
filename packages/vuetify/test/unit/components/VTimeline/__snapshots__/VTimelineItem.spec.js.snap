// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VTimelineItem.js should conditionally render an icon or icon slot 1`] = `

<div class="v-timeline-item theme--light">
  <div class="v-timeline-item__dot">
    <div class="v-timeline-item__inner-dot primary">
      <div>
        foo
      </div>
    </div>
  </div>
  <div class="v-timeline-item__body">
  </div>
</div>

`;

exports[`VTimelineItem.js should conditionally render an icon or icon slot 2`] = `

<div class="v-timeline-item theme--light">
  <div class="v-timeline-item__dot">
    <div class="v-timeline-item__inner-dot primary">
      <i aria-hidden="true"
         class="v-icon material-icons theme--dark"
      >
        foo
      </i>
    </div>
  </div>
  <div class="v-timeline-item__body">
  </div>
</div>

`;

exports[`VTimelineItem.js should conditionally render dot 1`] = `

<div class="v-timeline-item theme--light">
  <div class="v-timeline-item__body">
  </div>
</div>

`;

exports[`VTimelineItem.js should conditionally render dot 2`] = `

<div class="v-timeline-item theme--light">
  <div class="v-timeline-item__dot">
    <div class="v-timeline-item__inner-dot primary">
    </div>
  </div>
  <div class="v-timeline-item__body">
  </div>
</div>

`;

exports[`VTimelineItem.js should render a large icon correctly 1`] = `

<div class="v-timeline-item theme--light">
  <div class="v-timeline-item__dot v-timeline-item__dot--large">
    <div class="v-timeline-item__inner-dot primary">
    </div>
  </div>
  <div class="v-timeline-item__body">
  </div>
</div>

`;

exports[`VTimelineItem.js should render a medium icon correctly 1`] = `

<div class="v-timeline-item theme--light">
  <div class="v-timeline-item__dot v-timeline-item__dot--medium">
    <div class="v-timeline-item__inner-dot primary">
    </div>
  </div>
  <div class="v-timeline-item__body">
  </div>
</div>

`;

exports[`VTimelineItem.js should render a small icon correctly 1`] = `

<div class="v-timeline-item theme--light">
  <div class="v-timeline-item__dot v-timeline-item__dot--small">
    <div class="v-timeline-item__inner-dot primary">
    </div>
  </div>
  <div class="v-timeline-item__body">
  </div>
</div>

`;
